src/main/java/io/cryostat/net/web/http/api/v1/TargetRecordingsPostHandler.java

```diff
@@ -185,57 +159,34 @@ public void handleAuthenticated(RoutingContext ctx) throws Exception {
-                                                                        connection, d.getName()),
-                                                                webServer.getReportURL(
-                                                                        connection, d.getName()));
-                                                    } catch (QuantityConversionException
-                                                            | URISyntaxException
-                                                            | IOException e) {
-                                                        throw new HttpStatusException(500, e);
-                                                    }
-                                                });
+                                Pair<String, TemplateType> template =
+                                        RecordingCreationHelper.parseEventSpecifierToTemplate(
+                                                eventSpecifier);
+                                IRecordingDescriptor descriptor =
+                                        recordingCreationHelper.startRecording(
+                                                connectionDescriptor,
+                                                builder.build(),
+                                                template.getLeft(),
+                                                template.getRight());
+                                try {
+                                    WebServer webServer = webServerProvider.get();
+                                    return new HyperlinkedSerializableRecordingDescriptor(
+                                            descriptor,
+                                            webServer.getDownloadURL(
+                                                    connection, descriptor.getName()),
+                                            webServer.getReportURL(
+                                                    connection, descriptor.getName()));
+                                } catch (QuantityConversionException
+                                        | URISyntaxException
+                                        | IOException e) {
+                                    throw new HttpStatusException(500, e);
+                                }
                             });
 
-            descriptor.ifPresentOrElse(
-                    linkedDescriptor -> {
-                        ctx.response().setStatusCode(201);
-                        ctx.response().putHeader(HttpHeaders.LOCATION, "/" + recordingName);
-                        ctx.response()
-                                .putHeader(HttpHeaders.CONTENT_TYPE, HttpMimeType.JSON.mime());
-                        ctx.response().end(gson.toJson(linkedDescriptor));
-                    },
-                    () -> {
-                        throw new HttpStatusException(
-                                500, "Unexpected failure to create recording");
```

Comment:
This exception for when a recording fails unexpectedly seem to have been removed and hasn't been extracted out to the `RecordingCreationHelper` either. Is this because it isn't possible for a recording to fail for any reason other than those covered by the exceptions that were kept in?

Reply:
The `ifPresentOrElse` handling relates back to the simplification of `Optional` above. If the `Optional` was somehow (not sure in what scenarios it might realistically arise) empty, then this exception would be thrown to signify that unexpected failure. Without the `Optional` above, any unexpected exception will already become an HTTP 500 by the WebServer's failureHandler, so there is no need for an equivalent `throw` anymore.